scripts/cl/main.py

```diff
@@ -0,0 +1,267 @@
+        SqrtPriceRange(4545, None, fixed_prec_dec("670416215.718827443660400594")), # last one must be computed based on remaining token in, therefore it is None
+    ]
+
+    token_out_total, fee_growth_per_share_total = estimate_test_case(tick_ranges, token_in_initial, swap_fee, is_zero_for_one)
+
+    expected_token_out_total = fixed_prec_dec("8440821620.46523833169832895388")
+    expected_fee_growth_per_share_total = fixed_prec_dec("0.00000555275275702765744105956374059")
+
+    validate_confirmed_results(token_out_total, fee_growth_per_share_total, expected_token_out_total, expected_fee_growth_per_share_total)
+
+def estimate_consecutive_positions_gap_ofz_test():
+    """Estimates and prints the results of a calc concentrated liquidity test case with consecutive positions with a gap
+    when swapping token one for zero (ofz).
+
+     go test -timeout 30s -v -run TestKeeperTestSuite/TestCalcAndSwapOutAmtGivenIn/fee_6 github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity
+    """
+
+    is_zero_for_one = False
+    swap_fee = fixed_prec_dec("0.03")
+    token_in_initial = fixed_prec_dec("10000000000")
+
+    tick_ranges = [
+        SqrtPriceRange(5000, 5500, fixed_prec_dec("1517882343.751510418088349649")),
+        SqrtPriceRange(5501, None, fixed_prec_dec("1199528406.187413669220037261")), # last one must be computed based on remaining token in, therefore it is None
+    ]
+
+    token_out_total, fee_growth_per_share_total = estimate_test_case(tick_ranges, token_in_initial, swap_fee, is_zero_for_one)
+
+    expected_token_out_total = fixed_prec_dec("1772029.65214390801589935811000")
+    expected_fee_growth_per_share_total = fixed_prec_dec("0.218688507910948644574193665912")
+
+    validate_confirmed_results(token_out_total, fee_growth_per_share_total, expected_token_out_total, expected_fee_growth_per_share_total)
+
+def estimate_slippage_protection_zfo_test():
+    """Estimates and prints the results of a calc concentrated liquidity test case with slippage protection
+    when swapping token zero for one (zfo).
+
+     go test -timeout 30s -v -run TestKeeperTestSuite/TestCalcAndSwapOutAmtGivenIn/fee_7 github.com/osmosis-labs/osmosis/v14/x/concentrated-liquidity
+    """
+
+    is_zero_for_one = True
+    swap_fee = fixed_prec_dec("0.01")
+    token_in_initial = fixed_prec_dec("13370")
+
+    tick_ranges = [
+        SqrtPriceRange(5000, 4994, fixed_prec_dec("1517882343.751510418088349649")),
+    ]
+
+    token_out_total, fee_growth_per_share_total = estimate_test_case(tick_ranges, token_in_initial, swap_fee, is_zero_for_one)
+
+    expected_token_out_total = fixed_prec_dec("64417624.9871649525380486017974")
+    expected_fee_growth_per_share_total = fixed_prec_dec("0.0000000849292577225588233432564611676")
+
+    validate_confirmed_results(token_out_total, fee_growth_per_share_total, expected_token_out_total, expected_fee_growth_per_share_total)
+
+def main():
```

Comment:
Note: For scripts, I suggest reviewing this file in-detail and comparing it against Go test cases. However, I don't think there is value in reviewing other internal script calculations so feel free to skip